testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/security/loginmodules/policy/PolicyRegistrationTestCase.java

```diff
@@ -0,0 +1,104 @@
+import org.jboss.as.arquillian.api.ServerSetup;
+import org.jboss.as.test.integration.security.common.Utils;
+import org.jboss.shrinkwrap.api.Archive;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+
+/**
+ * Tests if NameNotFoundException due to policyRegistration occurs after logging into deployed secured ejb
+ * Test for [ JBEAP-13973 ].
+ *
+ * @author Daniel Cihak
+ */
+@RunWith(Arquillian.class)
+@ServerSetup({ SecurityTraceServerSetupTask.class })
+@RunAsClient
+public class PolicyRegistrationTestCase {
+
+    private static final String SECURED_EJB_DEPLOYMENT = "secured-ejb";
+
+    @Deployment(name = SECURED_EJB_DEPLOYMENT, testable = false)
+    public static Archive<?> createSecuredEjbDeployment() throws IOException {
+        WebArchive war = ShrinkWrap.create(WebArchive.class, SECURED_EJB_DEPLOYMENT + ".war");
+        war.addClasses(PolicyRegistrationTestCase.class, SecuredEJBServlet.class, SecuredEJB.class);
+        war.setWebXML(Utils.getResource("org/jboss/as/test/integration/security/loginmodules/policy/WEB-INF/web.xml"));
+        war.addAsResource(PolicyRegistrationTestCase.class.getPackage(), "jboss-ejb-client.properties", "jboss-ejb-client.properties");
+        ClassLoader tccl = Thread.currentThread().getContextClassLoader();
+        URL index = tccl.getResource("index.html");
+        war.addAsWebResource(index, "index.html");
+        return war;
+    }
+
+    /**
+     * Asserts the server.log file does not contain NameNotFoundException
+     *
+     * @throws IOException
+     */
+    @AfterClass
+    public static final void after() throws IOException {
+        File logFile = new File(SecurityTraceServerSetupTask.SERVER_LOG_DIR_VALUE, "server.log");
+        Assert.assertTrue("Log file " + logFile + " should exist", logFile.exists());
+        String logContent = new String(Files.readAllBytes(Paths.get(logFile.getPath())), StandardCharsets.UTF_8);
+        Assert.assertFalse(logContent.contains("javax.naming.NameNotFoundException: policyRegistration"));
+    }
+
+    @Test
+    @OperateOnDeployment(SECURED_EJB_DEPLOYMENT)
+    public void test(@ArquillianResource URL url) throws Exception {
+        HttpGet httpget = new HttpGet(url.toString() + "SecuredEJBServlet");
+        final String userpassword = "user1" + ":" + "password1";
+        final String headerValue = java.util.Base64.getEncoder().encodeToString(userpassword.getBytes());
+
+        Assert.assertNotNull(headerValue);
+        httpget.addHeader("Authorization", "Basic " + headerValue);
+
+        try (CloseableHttpClient httpclient = HttpClients.createDefault()) {
+            httpclient.execute(httpget);
```

Comment:
What are you actually testing here? Nothing here looks like it will fail the test, if the server responds with an error code the client does not throw an exception